File: API/Entities/Ad.cs
using System;
using System.Collections.Generic;
using HelpAFamilyOfferAChance.API.Entities;

namespace API.Entities
{
    public class Ad
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public double Quantity { get; set; }
        public bool ExistsLimit { get; set; }
        public double ? Limit { get; set; }
        public bool IsActive { get; set; }
        public User User{get;set;}
        public int UserId{get;set;}
        public UnitOfMeasure UnitOfMeasure{get;set;}
        public int UnitOfMeasureId{get;set;}
        public Category Category{get;set;}
        public int CategoryId{get;set;}
        public ICollection<Photo>  Photos {get;set;}
         public ICollection<Ad_x_DeliveryType>  Ad_x_DeliveryType{get;set;}
         public ICollection<Demand> Demands { get; set; }

    }}

File: front-end/src/app/models/ad.ts
import { Ad_x_DeliveryType } from "./ad_x_deliveryType";
import { Demand } from "./demand";
import { Photo } from "./photo";

export interface Ad
{
    id:number;
    name:string;
    description:string;
    createdAt:Date;
    quantity: number;
    existsLimit:boolean;
    limit?:number;
    isActive:boolean;
    url:string;
    userId:number;
    unitOfMeasure:string;
    category:string;
    unitOfMeasureId:number;
    categoryId:number;
    photos: Photo[];
    ad_x_DeliveryType: Ad_x_DeliveryType[];
    demands:Demand[];
}